server/service/teams.go

```diff
@@ -48,6 +49,42 @@ func (svc *Service) ListTeams(ctx context.Context, opt fleet.ListOptions) ([]*fl
 	return nil, fleet.ErrMissingLicense
 }
 
+func obfuscateSecrets(user *fleet.User, teams []fleet.Team) error {
+	if user == nil {
+		return &authz.Forbidden{}
+	}
+
+	isGlobalObs := user.IsGlobalObserver()
+	obsMembership := user.TeamObserverMembership()
+
+	for _, t := range teams {
+		if isGlobalObs || obsMembership[t.ID] {
+			for _, s := range t.Secrets {
+				s.Secret = "***"
+			}
+		}
+	}
+	return nil
+}
+
+func (r listTeamsResponse) hijackRender(ctx context.Context, w http.ResponseWriter) {
```

Comment:
I don't think we should be using `hijackRender` for this purpose:
```go
// renderHijacker can be implemented by response values to take control of
// their own rendering.
type renderHijacker interface {
	hijackRender(ctx context.Context, w http.ResponseWriter)
}
```
It's used to control of pages to render or to set special headers on the response.